app-android/app/src/main/java/de/tutao/tutanota/Native.kt

```diff
@@ -347,21 +359,23 @@ class Native internal constructor(
 			files.writeFile(logoFile, logoInputStream)
 			val logoUri = FileProvider.getUriForFile(activity, BuildConfig.FILE_PROVIDER_AUTHORITY, logoFile)
 			val thumbnail = ClipData.newUri(
-					activity.contentResolver,
-					"tutanota_logo",
-					logoUri
+				activity.contentResolver,
+				"tutanota_logo",
+				logoUri
 			)
 			sendIntent.clipData = thumbnail
 		} catch (e: IOException) {
-			Log.e(TAG, """
+			Log.e(
+				TAG, """
      Error attaching thumbnail to share intent:
      ${e.message}
-     """.trimIndent())
+     """.trimIndent()
+			)
 		}
 		sendIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
 		val intent = Intent.createChooser(sendIntent, null)
 		activity.startActivity(intent)
```

Comment:
you are running the whole function on `IO` dispatcher but this part should def be on `Main`